xarray/tests/test_dataarray.py

```diff
@@ -3024,6 +3024,47 @@ def test_to_pandas(self):
         with raises_regex(ValueError, 'cannot convert'):
             DataArray(np.random.randn(1, 2, 3, 4, 5)).to_pandas()
 
+    def test_index_immutability(self):
+        # Test for immutability of Pandas index from #2949
+        index = pd.Index(list('abcdefghij'), name='persistent')
+        series = pd.Series(np.random.randn(10), index)
+        array = xr.DataArray.from_series(series)
+        dataset = xr.Dataset({'myvar': array})
+
+        # Utility function that allows us to consistently many different paths
+        # to accessing the index.
+        def test_index(get_index):
+            # Test `index.data`
+            with pytest.raises(Exception):
+                get_index().data = np.random.randn(10)
+
+            # Test `index.dtype`
+            with pytest.raises(Exception):
+                get_index().dtype = np.float64
+
+            # Test `index.copy`
+            assert callable(get_index().copy)
+            get_index().copy = 'mutable?'
```

Comment:
I'm not sure we should be testing overriding a method. I don't think pandas would consider the ability to override `Index.copy` a guarantee of their public API, so this might break in the future, e.g., if pandas rewrote `Index` to be a CPython extension class rather than a class written in Python.

Reply:
I thinking was the same as [here](https://github.com/pydata/xarray/pull/3126#discussion_r304272667). If you'd rather limit the scope of this PR to guarantee that only `name` behaves as expected, I can remove these.